projects/core/src/auth/user-auth/events/user-auth-event.builder.ts

```diff
@@ -33,9 +42,17 @@ export class UserAuthEventBuilder {
    * Register a logout event
    */
   protected registerLogoutEvent(): void {
-    this.stateEventService.register({
-      action: AuthActions.LOGOUT,
-      event: LogoutEvent,
-    });
+    this.eventService.register(LogoutEvent, this.buildLogoutEvent());
+  }
+
+  /**
+   * Returns logout event stream
+   */
+  protected buildLogoutEvent(): Observable<LogoutEvent> {
+    return this.authService.isUserLoggedIn().pipe(
+      pairwise(),
+      filter(([prev, curr]) => prev && !curr),
+      map(() => createFrom(LogoutEvent, {}))
```

Comment:
We could just do `map(() => ({}))`, to save some bytes ;)

Reply:
Scratch this comment, let's have it as it is. 
We have a feature in events service (on TMS branch) for which my suggestion would cause some problems.